# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1550007885 -3600
#      Tue Feb 12 22:44:45 2019 +0100
# Branch stable
# Node ID b3b78bf8311340daa7f10442c188b2fe8f66dd63
# Parent  7b9a5ab8350fd0b584b0d4f14674e05dcf6b6aa5
Properly close figures after tests on hgload (bug #53468)

diff --git a/test/bug-53468/bug-53468.tst b/test/bug-53468/bug-53468.tst
--- a/test/bug-53468/bug-53468.tst
+++ b/test/bug-53468/bug-53468.tst
@@ -13,17 +13,35 @@
 ## GNU General Public License for more details.
 ##
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <https://www.gnu.org/licenses/>.
 
 ## Load an Octave 4.0.3 figure file
 %!test
-%! hgload ("ofig403.ofig");
+%! unwind_protect
+%!   hf = hgload ("ofig403.ofig");
+%! unwind_protect_cleanup
+%!   if (exist ("hf", "var") && isfigure (hf))
+%!     close (hf)
+%!   endif
+%! end_unwind_protect
 
 ## Load an Octave 4.2.1 figure file
 %!test
-%! hgload ("ofig421.ofig");
+%! unwind_protect
+%!   hf = hgload ("ofig421.ofig");
+%! unwind_protect_cleanup
+%!   if (exist ("hf", "var") && isfigure (hf))
+%!     close (hf)
+%!   endif
+%! end_unwind_protect
 
 ## Load an Octave 5.1 figure file
 %!test
-%! hgload ("ofig51.ofig");
+%! unwind_protect
+%!   hf = hgload ("ofig51.ofig");
+%! unwind_protect_cleanup
+%!   if (exist ("hf", "var") && isfigure (hf))
+%!     close (hf)
+%!   endif
+%! end_unwind_protect
